torch/linalg/__init__.py

```diff
@@ -367,6 +367,91 @@
             [-3.1113,  2.7381]], dtype=torch.float64)
 """)
 
+lstsq = _add_docstr(_linalg.linalg_lstsq, r"""
+torch.linalg.lstsq(input, b, cond=None, *, driver=None)
+    -> (Tensor solution, Tensor residuals, Tensor rank, Tensor singular_values)
+
+Computes the least squares solution to the system with a batch of matrices :math:`a` (represented by :attr:`input`)
+and a batch of vectors/matrices :math:`b` such that
+
+.. math::
+    x = \text{argmin}_x \|ax - b\|_F,
+
+where :math:`a` is of size :math:`(..., m, n)` and :math:`b` is of size :math:`(..., m, k)` or
+:math:`(..., m)`. The batch dimensions of :math:`a` and :math:`b` have to be broadcastable.
+
+The returned solution :math:`solution` is of shape :math:`(..., n, k)` if :math:`b` is of shape :math:`(..., m, k)`,
+and is of shape :math:`(..., n, 1)` if :math:`b` is of shape :math:`(..., m)`.
+The batch sizes of :math:`x` is the broadcasted shape of the batch dimensions of :math:`a` and :math:`b`.
+
+.. note::
+    The case when :math:`m < n` is not supported on the GPU yet.
+
+Args:
+    input (Tensor): the batch of left-hand side matrices :math:`a`
+        of shape :math:`(..., m, n)` with :math:`m > 0, n > 0`
+    b (Tensor): the batch of righ-hand side vectors or matrices :math:`b`
+        of shape :math:`(..., m)` or :math:`(..., m, k)` with :math:`m > 0, k > 0`
+    cond (float, optional): used to determine the effective rank of :math:`a`
+        for the rank-revealing drivers (see :attr:`driver`).
+        Singular values :math:`s[i] \le cond * s[0]` are treated as zero.
+        If :attr:`cond` is ``None`` or is smaller than zero,
+        the machine precision based on ``a.dtype`` is used.
```

Comment:
"based on \`\`a.dtype\`\` is used." -> "based on :attr:\`input\`'s dtype is used."